src/infra/datasets/postgres/src/repos/postgres_dataset_dependency_repository.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use database_common::{TransactionRef, TransactionRefT};
+use dill::{component, interface};
+use internal_error::{ErrorIntoInternal, InternalError, ResultIntoInternal};
+use kamu_datasets::*;
+use opendatafabric::DatasetID;
+use sqlx::{Postgres, QueryBuilder};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct PostgresDatasetDependencyRepository {
+    transaction: TransactionRefT<sqlx::Postgres>,
+}
+
+#[component(pub)]
+#[interface(dyn DatasetDependencyRepository)]
+impl PostgresDatasetDependencyRepository {
+    pub fn new(transaction: TransactionRef) -> Self {
+        Self {
+            transaction: transaction.into(),
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl DatasetDependencyRepository for PostgresDatasetDependencyRepository {
+    async fn stores_any_dependencies(&self) -> Result<bool, InternalError> {
+        let mut tr = self.transaction.lock().await;
+
+        let connection_mut = tr.connection_mut().await?;
+
+        let has_data = sqlx::query_scalar!(
+            r#"
+            SELECT EXISTS (SELECT * FROM dataset_dependencies LIMIT 1) as has_data
```

Comment:
Q: out of curiosity, why not just `COUNT(*)`?

Reply:
I assume this query will at most look at 1 row, while `COUNT` is likely doing more work. Even if it's not loading all rows, it's still sort of a computation over table pages.

While we only need to know if table is empty...

Reply:
Yeah, makes a lot of sense -- thank you